script/dump_stats.rb

```diff
@@ -0,0 +1,116 @@
+# 1. Run this script on production as follows:
+#      RAILS_ENV=production bundle exec rails r script/dump_stats.rb
+# frozen_string_literal: true
+
+work_output = File.open(ENV['WORKS_FILENAME'] || '/tmp/works.csv', 'w')
+work_output.puts "Created_at,language,restricted,complete,word_count,tags\n"
+if ENV['QUICK'].nil?
+  stats_output = File.open(ENV['STATS_FILENAME'] || '/tmp/stats.csv', 'w')
+  stats_output.puts "count_of_days,to,from,day_of_week,tag_type,tag_ids,count"
+end
+first_time = Work.first.created_at
+days = 1
+end_date = DateTime.now.beginning_of_day
+start_date = end_date - 1.day
+loop do
+  w = Work.where(posted: true, created_at: start_date..end_date)
+  fandoms = {}
+  fandoms.default = 0
+  categories = {}
+  categories.default = 0
+  ratings = {}
+  ratings.default = 0
+  warnings = {}
+  warnings.default = 0
+  characters = {}
+  characters.default = 0
+  relationships = {}
+  relationships.default = 0
+  freeforms = {}
+  freeforms.default = 0
+  warnings_multi = {}
+  warnings_multi.default = 0
+  categories_multi = {}
+  categories_multi.default = 0
+  w.each do |work|
+    work_output.puts "#{work.created_at.strftime('%F')},#{work&.language&.short},#{work&.restricted},#{work&.complete},#{work&.word_count},#{work&.tags&.pluck(:id)&.join('+')}\n"
```

Comment:
I don't think you need safe navigation on the work here. You should be able to replace `work&.` with `work.` throughout the loop. (In fact, I think the only place you need safe navigation in this loop is for `work.language&.short`.)